verilog/formatting/formatter.cc

```diff
@@ -195,66 +195,262 @@ Status FormatVerilog(absl::string_view text, absl::string_view filename,
     const TokenPartitionTree& format_tokens_partitions,
     const FormatStyle& style) {
-  // Initialize a tree view that treats partitions as fully-expanded.
-  ExpandableTreeView<UnwrappedLine> format_tokens_partition_view(
-      format_tokens_partitions);
 
-  // For unwrapped lines that fit, don't bother expanding their partitions.
-  // Post-order traversal: if a child doesn't 'fit' and needs to be expanded,
-  // so must all of its parents (and transitively, ancestors).
+  ProxyTreeView<UnwrappedLine> format_tokens_partition_view(format_tokens_partitions);
+
+  // Populate nodes
   format_tokens_partition_view.ApplyPostOrder(
-      [&style](partition_node_type& node) {
-        DeterminePartitionExpansion(&node, style);
+      [&style](VectorTree<ProxyTreeViewNodeInfo<UnwrappedLine>>& node) {
+        auto& node_view = node.Value();
+        auto& uw = node_view.Value();
+
+        if (node.Children().size() == 0) {
+          node_view.unwrapped_lines.push_back(uw);
+          return ;
+        }
+
+        switch (uw.PartitionPolicy()) {
+        default:
+        case PartitionPolicyEnum::kAlwaysExpand: {
+          for (const auto& itr : node.Children()) {
+            std::copy(itr.Value().unwrapped_lines.begin(),
+                      itr.Value().unwrapped_lines.end(),
+                      std::back_inserter(node_view.unwrapped_lines));
+          }
+          break;
+        }
+
+        case PartitionPolicyEnum::kFitOnLineElseExpand: {
+          std::vector<UnwrappedLine> unwrapped_lines;
+
+          for (auto& itr : node.Children()) {
+            std::copy(itr.Value().unwrapped_lines.begin(),
+                      itr.Value().unwrapped_lines.end(),
+                      std::back_inserter(unwrapped_lines));
+          }
+
+          // fit everyting in one line
+          UnwrappedLine uwline(unwrapped_lines.begin()->IndentationSpaces(),
+                               unwrapped_lines.begin()->TokensRange().begin());
+          uwline.SpanUpToToken(unwrapped_lines.back().TokensRange().end());
+
+          if (verible::FitsOnLine(uwline, style).fits) {
+            node_view.unwrapped_lines.push_back(uwline);
+          } else {
+            std::copy(unwrapped_lines.begin(),
+                      unwrapped_lines.end(),
+                      std::back_inserter(node_view.unwrapped_lines));
+          }
+          break;
+        }
+
+        case PartitionPolicyEnum::kAppendFittingSubPartitions: {
+          std::vector<UnwrappedLine> unwrapped_lines;
+
+          for (auto& itr : node.Children()) {
+            std::copy(itr.Value().unwrapped_lines.begin(),
+                      itr.Value().unwrapped_lines.end(),
+                      std::back_inserter(unwrapped_lines));
+          }
+
+          const auto& unwrapped = MakeUnwrappedLines(unwrapped_lines, style, false);
+          const auto& wrapped   = MakeUnwrappedLines(unwrapped_lines, style, true);
+
+          if (unwrapped.size() <= wrapped.size()) {
```

Comment:
Just to confirm: Is this a comparison of the number of lines of the above results?